src/sunlinsol/spbcgs/sunlinsol_spbcgs.c

```diff
@@ -240,135 +199,130 @@ SUNLinearSolver_ID SUNLinSolGetID_SPBCGS(SUNLinearSolver S)
   return (SUNLINEARSOLVER_SPBCGS);
 }
 
-int SUNLinSolInitialize_SPBCGS(SUNLinearSolver S)
+SUNErrCode SUNLinSolInitialize_SPBCGS(SUNLinearSolver S)
 {
-  /* ensure valid options */
-  if (S == NULL) { return (SUNLS_MEM_NULL); }
+  SUNFunctionBegin(S->sunctx);
 
   if (SPBCGS_CONTENT(S)->maxl <= 0)
   {
     SPBCGS_CONTENT(S)->maxl = SUNSPBCGS_MAXL_DEFAULT;
   }
 
-  if (SPBCGS_CONTENT(S)->ATimes == NULL)
-  {
-    LASTFLAG(S) = SUNLS_ATIMES_NULL;
-    return (LASTFLAG(S));
-  }
+  SUNAssert(SPBCGS_CONTENT(S)->ATimes, SUN_ERR_ARG_CORRUPT);
 
   if ((PRETYPE(S) != SUN_PREC_LEFT) && (PRETYPE(S) != SUN_PREC_RIGHT) &&
       (PRETYPE(S) != SUN_PREC_BOTH))
   {
     PRETYPE(S) = SUN_PREC_NONE;
   }
 
-  if ((PRETYPE(S) != SUN_PREC_NONE) && (SPBCGS_CONTENT(S)->Psolve == NULL))
-  {
-    LASTFLAG(S) = SUNLS_PSOLVE_NULL;
-    return (LASTFLAG(S));
-  }
+  SUNAssert((PRETYPE(S) == SUN_PREC_NONE) || SPBCGS_CONTENT(S)->Psolve,
+            SUN_ERR_ARG_CORRUPT);
 
   /* no additional memory to allocate */
 
   /* return with success */
-  LASTFLAG(S) = SUNLS_SUCCESS;
```

Comment:
Here and elsewhere, should we still set `LASTFLAG`?

Reply:
At the Tuesday meeting, we determined that we woul will only set `LASTFLAG` in Setup/Solve. I think it needs to be set in `Initialize` too. However, I have removed it from the Set functions.